src/DataSet/AttributeDataSet.php

```diff
@@ -36,82 +31,19 @@ public function getRules(): iterable
     {
         $classMeta = new ReflectionClass($this->baseAnnotatedObject);
 
-        return $this->handleAttributes($classMeta);
+        return $this->collectAttributes($classMeta);
     }
 
-    private function handleAttributes(ReflectionClass $classMeta): array
+    // TODO: use Generator to collect attributes
+    private function collectAttributes(ReflectionClass $classMeta): array
     {
         $rules = [];
         foreach ($classMeta->getProperties() as $property) {
-            if ($property->isStatic()) {
-                continue;
-            }
-
-            foreach ([HasMany::class, HasOne::class] as $className) {
-                $attributes = $property->getAttributes($className);
-                if (empty($attributes)) {
-                    continue;
-                }
-
-                $relatedClassName = $attributes[0]->getArguments()[0];
-
-                try {
-                    $relatedClassMeta = new ReflectionClass($relatedClassName);
-                } catch (ReflectionException) {
-                    throw new InvalidArgumentException("Class \"$relatedClassName\" not found.");
-                }
-
-                $nestedRule = new Nested(
-                    $this->handleAttributes($relatedClassMeta),
-                    ...(($property->getAttributes(Nested::class)[0] ?? null)?->getArguments() ?? [])
-                );
-
-                if ($className !== HasMany::class) {
-                    $rules[$property->getName()] = $nestedRule;
-                } else {
-                    /** @psalm-suppress UndefinedMethod */
-                    $rules[$property->getName()][] = new Each(
-                        [$nestedRule],
-                        ...(($property->getAttributes(Each::class)[0] ?? null)?->getArguments() ?? [])
-                    );
-                }
-            }
-
-            $eachRuleFound = false;
-            $eachRules = [];
-            $attributes = $property->getAttributes();
+            $attributes = $property->getAttributes(RuleInterface::class, ReflectionAttribute::IS_INSTANCEOF);
```

Comment:
Now we collect static properties as well?

Reply:
yep